red-hat-data-services/odh-manifests
# Switch to manual prometheus operator deployment.

Instead of relying on OLM to provision the prometheus operator, we are opting to manually deploy the prometheus operator. This reduces a lot of overhead involved with adding a new olm dependency within rhods. It will also give us a lot more flexibility in the sort of namespaces the prometheus operator has access to, without conflicting with existing ocp monitoring stack.

I've also reorganized the kustomize manifests to separate out each resource in its own file, as this is more consistent with how manifests are organized in the rest of the repository.


### Testing: 

Live build: quay.io/hukhan/rhods-operator-live-catalog:1.20.2-6053

Steps to test:
* `git clone https://gitlab.cee.redhat.com/data-hub/olminstall.git && cd olminstall`
* `./setup.sh -t operator  -i quay.io/hukhan/rhods-operator-live-catalog:1.20.2-6053`
* Wait for the modelmesh monitoring stack pods to come up
* Confirm the prometheus operator comes up succesfully, and prometheus instance is deployed 
  * `oc get pods redhat-ods-monitoring` 
* ` git clone https://github.com/opendatahub-io/modelmesh-serving.git && cd modelmesh-serving/quickstart`
* Comment out lines 32 and 33 (do not need to deploy MM, rhods operator will do this for us)
* ` ./quickstart.sh` 
* Set up port forwarding for the new monitoring pod: 
  * `oc port-forward prometheus-rhods-model-monitoring-0 -n redhat-ods-monitoring 9090`
* Connect to localhost:9090 on your browser. You will see the new targets
* Confirm you see modelmesh and haproxy metrics. 

### Notes
* You may see some errors logs in the operator pod stating `failed to watch x ... connection refused`, these errors existed in the olm installation as well, and is not unique to the manual deployment method 
* You may see errors in the prometheus instance "Error on ingesting out-of-order samples", these also appeared in the prometheus deployed via olm in my testing, so it is not related to the deployment method


- [x] JIRA link(s): https://issues.redhat.com/browse/RHODS-6053
